lib/registrant.rb

```diff
@@ -0,0 +1,35 @@
+require "./spec/spec_helper"
+
+# The Registrant class represents a single person and their license data.
+# This includes their name, age, and whether they have a driving permit,
+# driver's license, and whether their license has been renewed.
+# Syntax:
+#
+# registrant = Registrant.new({name: "Bill", age: 18})
+#
+# Methods:
+#
+# registrant.earn_permit
+# registrant.permit?
+class Registrant
+  attr_reader :name, :age, :permit
+  attr_accessor :license_data
+
+  def initialize(info)
+    @name = info[:name]
+    @age = info[:age]
+    @permit = info[:permit] || false
+    @license_data = { written: false, license: false, renewed: false }
+    return if info[:license_data].nil?
```

Comment:
hmm, I'm not sure why you have this guard clause. Line 22 sets the license_data for us so we don't need to look into the arguments for this data.